readers/hdfs/src/main/java/com/criteo/hadoop/garmadon/hdfs/monitoring/PrometheusMetrics.java

```diff
@@ -4,51 +4,138 @@
-                String.valueOf(partition));
+    public static Counter.Child tmpFileOpenFailuresCounter(String eventName) {
+        return buildChild(TMP_FILE_OPEN_FAILURES, eventName);
+    }
+
+    public static Counter.Child tmpFilesOpened(String eventName) {
+        return buildChild(TMP_FILES_OPENED, eventName);
+    }
+
+    public static Counter.Child filesCommittedCounter(String eventName) {
+        return buildChild(FILES_COMMITTED, eventName);
+    }
+
+    public static Counter.Child checkPointFailuresCounter(String eventName, int partition) {
+        registerPartitionCollector(CHECKPOINTS_FAILURES, eventName, String.valueOf(partition));
+        return buildChild(CHECKPOINTS_FAILURES, eventName, String.valueOf(partition));
+    }
+
+    public static Counter.Child checkPointSuccessesCounter(String eventName, int partition) {
+        registerPartitionCollector(CHECKPOINTS_SUCCESSES, eventName, String.valueOf(partition));
+        return buildChild(CHECKPOINTS_SUCCESSES, eventName, String.valueOf(partition));
+    }
+
+    public static Counter.Child messageWritingFailuresCounter(String eventName, int partition) {
+        registerPartitionCollector(MESSAGES_WRITING_FAILURES, eventName, String.valueOf(partition));
+        return buildChild(MESSAGES_WRITING_FAILURES, eventName, String.valueOf(partition));
+    }
+
+    public static Counter.Child messageWrittenCounter(String eventName, int partition) {
+        registerPartitionCollector(MESSAGES_WRITTEN, eventName, String.valueOf(partition));
+        return buildChild(MESSAGES_WRITTEN, eventName, String.valueOf(partition));
+    }
+
+    public static Counter.Child hearbeatsSentCounter(String eventName, int partition) {
+        registerPartitionCollector(HEARTBEATS_SENT, eventName, String.valueOf(partition));
+        return buildChild(HEARTBEATS_SENT, eventName, String.valueOf(partition));
+    }
+
+    public static Gauge.Child currentRunningOffsetsGauge(String eventName, int partition) {
+        registerPartitionCollector(CURRENT_RUNNING_OFFSETS, eventName, String.valueOf(partition));
+        return buildChild(CURRENT_RUNNING_OFFSETS, eventName, String.valueOf(partition));
+    }
+
+    public static Gauge.Child latestCommittedOffsetGauge(String eventName, int partition) {
+        registerPartitionCollector(LATEST_COMMITTED_OFFSETS, eventName, String.valueOf(partition));
+        return buildChild(LATEST_COMMITTED_OFFSETS, eventName, String.valueOf(partition));
+    }
+
+    public static Gauge.Child latestCommittedTimestampGauge(String eventName, int partition) {
+        registerPartitionCollector(LATEST_COMMITTED_TIMESTAMPS, eventName, String.valueOf(partition));
+        Gauge.Child gauge = buildChild(LATEST_COMMITTED_TIMESTAMPS, eventName, String.valueOf(partition));
+        return new Gauge.Child() {
+            @Override
+            public void set(double val) {
+                gauge.set(Double.max(gauge.get(), val));
```

Comment:
it can happen because commit is for every file. And their can be many files for one partition handled by an hdfs reader instance at the same moment -> possibly one per day (in the case of late events in kafka). If you only set the gauge, it is possible that you display the timestamp of a file written for the day before.
Another way of tackling it would be to add a tag 'day' in the prometheus metric and then take the max without day in prometheus alerting.
That creates more timeseries in prometheus but considering the lifetime of a prometheus instance that should not be a burden. 
let's discuss IRL

Reply:
OK